src/commands/build/upload.rs

```diff
@@ -172,7 +173,28 @@ pub fn execute(matches: &ArgMatches) -> Result<()> {
 
     let base_repo_name = matches.get_one("base_repo_name").map(String::as_str);
     let base_sha = matches.get_one("base_sha").map(String::as_str);
-    let base_ref = matches.get_one("base_ref").map(String::as_str);
+    let base_ref = matches
```

Comment:
I'd recommend moving these up into the `let` block where we get `vcs_provider`, `head_repo_name` and `head_ref` (line 117 here). That way we get some safe error handling and can re-use the repo and other variables